models/Search/Backend/Data.php

```diff
@@ -507,6 +510,28 @@ protected function cleanupData($data)
         $data = str_replace("\t", '', $data);
         $data = preg_replace('#[ ]+#', ' ', $data);
 
+        $minWordLength = $this->getDao()->getMinWordLengthForFulltextIndex();
+        $maxWordLength = $this->getDao()->getMaxWordLengthForFulltextIndex();
+
+        $words = explode(' ', $data);
+
+        $wordOccurrences = [];
+        foreach($words as $key => &$word) {
```

Comment:
Since you don't write `$word` in the loop, there's no need in assigning it as a reference.

Reply:
Is only to not create a copy of the array - saves memory.

Reply:
`foreach` loops in PHP are copy-on-write ;)

Reply:
Thanks, you are right ;-)
```
It re-uses the existing value in memory. With zero performance cost. It's called "copy-on-write" and means that PHP doesn't make any copies unless you try to MODIFY the value.
```
https://www.php.net/manual/de/control-structures.foreach.php#120710